sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
 """
-This submodule provides utility functions to act on `sunpy.map.Map` instances.
+This submodule provides utility functions to act on `sunpy.map.GenericMap` instances.
 """
+from itertools import chain, product
 
+import numpy as np
 
-__all__ = []
+import astropy.units as u
+from astropy.coordinates import SkyCoord
 
+from sunpy.coordinates import Helioprojective
+
+__all__ = ['all_pixel_indices_from_map', 'all_coordinates_from_map',
+           'map_edges', 'contains_full_disk',
+           'is_all_off_disk', 'is_all_on_disk', 'contains_limb',
+           'coordinate_is_on_disk', 'on_disk_bounding_coordinates']
+
+
+def all_pixel_indices_from_map(smap):
+    """
+    Returns pixel pair indices of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
```

Comment:
```suggestion
        A  `~sunpy.map.Map`.
```

Reply:
I am not sure how this renders? I don't think we have autodoc for `sunpy.map.Map` which is why I generally use `GenericMap` when referring to the type. I think if you want to refer to the factory you need to do `sunpy.map.map_factory.MapFactory`